.github/data/matrix-regression.json

```diff
@@ -0,0 +1,18 @@
+{
+  "k8s": [
+    "1.22.17",
```

Comment:
this needs to be updated from @j1m-ryan's recent changes

Reply:
I don't mind if we have the versions in code declaritvely like this in json. One advantage of this way is that we can probably automate that part of the release where we say what versions we tested it on, a little easier that if we are getting the versions from an API. But yeah we probably ought to add in the latest two verions here, 1.28.0 and 1.29.0

Reply:
@vepatel @j1m-ryan that should pull the versions dynamically now, from a local run the generated matrix was:

```json
{
  "k8s": [
    [
      "1.23.17",
      "1.24.15",
      "1.25.11",
      "1.26.6",
      "1.27.3",
      "1.28.0",
      "1.29.0"
    ]
  ],
  "images": [
    {
      "image": "debian"
    },
    {
      "image": "debian-plus"
    }
  ]
}
```

Reply:
Working matrix from manual regression trigger with @pdabelf5:
![image](https://github.com/nginxinc/kubernetes-ingress/assets/13047589/a04ff2c6-cbad-49a3-9fc2-60ca6fc3864e)